ml-proto/src/spec/arithmetic.ml

```diff
@@ -119,10 +123,30 @@ struct
 
   let unop op =
     let f = match op with
-      | Clz -> fun i -> i  (* TODO *)
-      | Ctz -> fun i -> i  (* TODO *)
-      | Popcnt -> fun i -> i  (* TODO *)
-    in fun v -> Int.to_value (f (Int.of_value 1 v))
+      | Clz -> let rec loop acc n =
```

Comment:
Also, how about adding a `let open Int in` here to avoid all the repetition? (similarly for the other cases)